src/DatePicker.elm

```diff
@@ -358,16 +360,22 @@ update settings msg (DatePicker ({ forceOpen, focused } as model)) =
                 False ->
                     let
                         dateEvent =
-                            case settings.parser <| Maybe.withDefault "" model.inputText of
-                                Ok date ->
-                                    if settings.isDisabled date then
-                                        FailedInput <| Disabled date
+                            case Maybe.withDefault "" model.inputText of
```

Comment:
I think it would be a bit more readable here, if the case split was directly on `model.inputText`
```elm
case model.inputText of
    Nothing -> FailedInput EmptyString
    Just "" -> FailedInput EmptyString
    Just a -> handle parsing
```